sdk/python/feast/data_source.py

```diff
@@ -512,6 +512,77 @@ def source_datatype_to_feast_value_type() -> Callable[[str], ValueType]:
         return type_map.redshift_to_feast_value_type
 
 
+class RequestDataSource(DataSource):
+    """
+    RequestDataSource that can be used to provide input features for on demand transforms
+
+    Args:
+        name: Name of the request data source
+        schema: Schema mapping from the input feature name to a ValueType
+    """
+
+    @staticmethod
+    def source_datatype_to_feast_value_type() -> Callable[[str], ValueType]:
+        pass
+
+    def get_table_query_string(self) -> str:
+        pass
```

Comment:
This shuld probably have a return type of `Optional[str]` for correctness.

Reply:
Just removed this since it's optional. Made the other method raise an exception